web/src/main/webapp/WEB-INF/pages/application_state.jsp

```diff
@@ -106,24 +134,90 @@
+                              <p class="form-control-static <c:if test="${stat.first}">stateData</c:if>" id="languageSkillP">${data.languageSkill}</p>
                               <c:if test="${stat.first}">
-                                  <input class="stateInput" type="text" name="languageSkill" id="languageSkillInput" style="display:none" value="${data.languageSkill}">
+                                  <input class="stateInput" type="number" name="languageSkill" id="languageSkillInput" style="display:none" value="${data.languageSkill}"
+                                  data-error="<spring:message code="candidate.error.language.incorrect"/>"
+                                  data-bind="valueWithInit: 'languageSkill'"
+                                  max="10" min="0">
                               </c:if>
                           </div>
+                          <div class="help-block with-errors"></div>
                       </div>
                   </c:when>
                   <c:when test="${data.stateName == 'firstTest'}">
                       <div class="form-group">
-                          <label for="name" class="control-label col-sm-4"><spring:message code="statehistory.field.result"/></label>
+                          <label for="resultInput" class="control-label col-sm-4"><spring:message code="statehistory.field.result"/></label>
                           <div class="col-sm-8">
-                              <p class="form-control-static <c:if test="${stat.first}">stateData"</c:if> id="resultP" >${data.result}</p>
+                              <p class="form-control-static <c:if test='${stat.first}'>stateData</c:if>" id="resultP" >${data.result}%</p>
                               <c:if test="${stat.first}">
-                                  <input class="stateInput" type="text" name="result" id="resultInput" style="display:none" value="${data.result}">
+                                  <input required class="stateInput" type="number" name="result" id="resultInput" style="display:none;" value="${data.result}"
+                                    data-bind="valueWithInit: 'result'"
+                                    max="100" min="0"
+                                    data-error="<spring:message code='statehistory.error.result.range'/>" />
                               </c:if>
                           </div>
+                          <div class="help-block with-errors"></div>
+                      </div>
+                      <div class="form-group">
+                        <label for="recommendationInput" class="control-label col-sm-4"><spring:message code="statehistory.field.recommendation"/></label>
+                        <div class="col-sm-8">
+                          <p class="form-control-static <c:if test="${stat.first}">stateData</c:if>" id="recommendationP">
```

Comment:
Rewrite the inner <c:if> structure to JSP EL trinary conditional operator, aka ?:

See http://docs.oracle.com/javaee/7/tutorial/jsf-el005.htm#BNAIK